### DISCUSSION TITLE
ros implementation

### DISCUSSION THREAD

**Discussion by grotius-cnc** [grotius-cnc/hal-core#5] [Ideas]
This is a test for discussion.

**Comment by jjrbfi** [comment]
The ROS implementation will be the next once I have the EtherCAT working with HAL. :) 

Will be based probably on [https://github.com/orocos/rtt_ros_integration](https://github.com/orocos/rtt_ros_integration)

**Reply by grotius-cnc** [reply]
How close are you to get ethercat running?
What are the real benefits using ros?

**Reply by grotius-cnc** [reply]
P.s. you have write permissions on this repository. I think that's handy.

**Reply by grotius-cnc** [reply]
We can add your compile procedure, to compile everything as seperated for ethercat.

**Reply by jjrbfi** [reply]
How close are you to get ethercat running?
    - Right now I'm trying to following your new changes and try to publish them and test but you are so fast 😂 .
    
What are the real benefits using ros?
    - Well, right now quite many researches and companies are using ROS for Robots. I tried months a go https://github.com
    
/orocos/rtt_soem but there is quite old. But I would like to get working with HAL because we can get more people interested in the HAL you have made and plus ROS will be a hype for sure. In LinuxCNC are quite many drivers working, so that's great.
    
P.s. you have write permissions on this repository. I think that's handy.
    - Thanks I will anyways try to be in contact with you here before doing "hard changes to files".

That is a good idea.
I would like to get working EtherCAT. Right now I'm facing some issues with it. I will make a new Discussion for it. Or you prefer to use LinuxCNC forum?
